src/jitlayers.cpp

```diff
@@ -79,14 +79,20 @@ extern "C" JL_DLLEXPORT
 uint64_t jl_cumulative_compile_time_ns_before()
 {
     int tid = jl_threadid();
-    jl_measure_compile_time[tid] = 1;
+    jl_measure_compile_time[tid] += 1;
```

Comment:
This has me worried in the context of migration. We can no longer assume that the Julia task will finish executing on the same thread that it started on. This would mean increment and decrement no longer matching. 

In https://github.com/JuliaLang/julia/pull/41612 we solved this by having a single atomic counter that keeps track of how many tasks have asked for timings to be enabled.
Similarily for the `jl_cumulative_compile_time` array.

Reply:
Interesting. So that's a global metric, not a per-thread metric.

In fact, doesn't task-migrations on 1.7 mean that `@time` is already broken on 1.7? Since you might migrate to a thread that isn't currently measuring compilation? 😬

Reply:
Oh, yeah, okay this is definitely just broken on 1.7:
https://github.com/JuliaLang/julia/issues/41739

**Sigh.**

I just wanted to make this small fix, why is everything so complicated!? 😂 

-----------

............ okay i'll think a bit more about doing this the Right Way.

Am I right that this PR as-is looks okay for a backport to 1.6 (for our own buildfarm, even if we don't _actually_ backport this to 1.6?)

Reply:
I've branched this into https://github.com/NHDaly/julia/tree/nhd-allow-reentrant-compilation-time-measurement--1.6-backport so that we can look at backporting that, separately.

Reply:
🤔 part of the problem is that this boolean is checked from C code. Is there an equivalent to atomics for our C codebase?

Reply:
EDIT: ah, yeah, okay i think i've found stuff after looking through the codebase... i'll try this

Reply:
Okay thanks, Valentin. I've addressed this in the latest push! :)